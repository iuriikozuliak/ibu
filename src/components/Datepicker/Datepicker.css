.Datepicker {

}

:global(.DateRangePicker) {
  width: 100%;
}

:global(.DateRangePickerInput) {
  width: 100%;
}
