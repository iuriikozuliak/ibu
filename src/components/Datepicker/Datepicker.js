import React, { Component } from 'react';
import { DateRangePicker }  from 'react-dates';

import 'react-dates/lib/css/_datepicker.css';

class Datepicker extends Component {
  constructor(props) {
    super(props);

    this.state = {
      focusedInput: null
    };

    this.onDatesChange = this.onDatesChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
  }

  onDatesChange({ startDate, endDate }) {
    return this.props.onChange([{
      name: 'startDate',
      value: startDate
    }, {
      name: 'endDate',
      value: endDate
    }]);
  }

  onFocusChange(focusedInput) {
    this.setState({ focusedInput });
  }

  render() {
    const {
      focusedInput
    } = this.state;

    const {
      startDate,
      endDate
    } = this.props;

    return (
      <DateRangePicker
        onDatesChange = { this.onDatesChange }
        onFocusChange = { this.onFocusChange }
        focusedInput  = { focusedInput }
        startDate     = { startDate }
        endDate       = { endDate }
        withPortal
        hideKeyboardShortcutsPanel />
    );
  }
}


export default Datepicker;
