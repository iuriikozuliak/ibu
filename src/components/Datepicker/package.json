{
  "name": "Datepicker",
  "private": true,
  "main": "./Datepicker.js"
}
