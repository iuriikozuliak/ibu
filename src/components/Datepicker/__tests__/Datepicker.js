import React       from 'react';
import renderer    from 'react-test-renderer';
import moment      from 'moment';
import { mount } from 'enzyme';

import Datepicker from '../Datepicker';

test('Renders Datepicker component', () => {
  const component = renderer.create(
    <Datepicker />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Renders Datepicker component with start and end dates', () => {
  const component = renderer.create(
    <Datepicker
      startDate = { moment('1995-12-25') }
      endDate   = { moment('1995-12-30') } />
  ).toJSON();
  expect(component).toMatchSnapshot();
});

test('Triggers onChange event on onDatesChange with correct payload', () => {
  let payload;
  const expectedPayload = [{
    name: 'startDate',
    value: moment('1995-12-25')
  }, {
    name: 'endDate',
    value: null
  }];
  const component = mount(
    <Datepicker
      startDate = { moment('1995-12-25') }
      endDate   = { null }
      onChange  = { v => payload = v } />
  );

  component.find('#startDate').simulate('change');
  component.find('#endDate').simulate('change');
  expect(payload).toEqual(expectedPayload);
});
