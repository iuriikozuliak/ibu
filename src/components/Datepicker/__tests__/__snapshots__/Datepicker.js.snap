// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Datepicker component 1`] = `
<div
  className="DateRangePicker"
>
  <div>
    <div
      className="DateRangePickerInput"
    >
      <div
        className="DateInput"
      >
        <input
          aria-describedby="DateInput__screen-reader-message-startDate"
          aria-label="Start Date"
          autoComplete="off"
          className="DateInput__input needsclick"
          disabled={false}
          id="startDate"
          name="startDate"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Start Date"
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
        <p
          className="screen-reader-only"
          id="DateInput__screen-reader-message-startDate"
        >
          Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
        </p>
        <div
          className="DateInput__display-text"
        >
          Start Date
        </div>
      </div>
      <div
        aria-hidden="true"
        className="DateRangePickerInput__arrow"
        role="presentation"
      >
        <svg
          viewBox="0 0 1000 1000"
        >
          <path
            d="M694.4 242.4l249.1 249.1c11 11 11 21 0 32L694.4 772.7c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210.1-210.1H67.1c-13 0-23-10-23-23s10-23 23-23h805.4L662.4 274.5c-21-21.1 11-53.1 32-32.1z"
          />
        </svg>
      </div>
      <div
        className="DateInput"
      >
        <input
          aria-describedby="DateInput__screen-reader-message-endDate"
          aria-label="End Date"
          autoComplete="off"
          className="DateInput__input needsclick"
          disabled={false}
          id="endDate"
          name="endDate"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="End Date"
          readOnly={false}
          required={false}
          type="text"
          value=""
        />
        <p
          className="screen-reader-only"
          id="DateInput__screen-reader-message-endDate"
        >
          Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
        </p>
        <div
          className="DateInput__display-text"
        >
          End Date
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Renders Datepicker component with start and end dates 1`] = `
<div
  className="DateRangePicker"
>
  <div>
    <div
      className="DateRangePickerInput"
    >
      <div
        className="DateInput"
      >
        <input
          aria-describedby="DateInput__screen-reader-message-startDate"
          aria-label="Start Date"
          autoComplete="off"
          className="DateInput__input needsclick"
          disabled={false}
          id="startDate"
          name="startDate"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Start Date"
          readOnly={false}
          required={false}
          type="text"
          value="1995-12-25"
        />
        <p
          className="screen-reader-only"
          id="DateInput__screen-reader-message-startDate"
        >
          Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
        </p>
        <div
          className="DateInput__display-text DateInput__display-text--has-input"
        >
          12/25/1995
        </div>
      </div>
      <div
        aria-hidden="true"
        className="DateRangePickerInput__arrow"
        role="presentation"
      >
        <svg
          viewBox="0 0 1000 1000"
        >
          <path
            d="M694.4 242.4l249.1 249.1c11 11 11 21 0 32L694.4 772.7c-5 5-10 7-16 7s-11-2-16-7c-11-11-11-21 0-32l210.1-210.1H67.1c-13 0-23-10-23-23s10-23 23-23h805.4L662.4 274.5c-21-21.1 11-53.1 32-32.1z"
          />
        </svg>
      </div>
      <div
        className="DateInput"
      >
        <input
          aria-describedby="DateInput__screen-reader-message-endDate"
          aria-label="End Date"
          autoComplete="off"
          className="DateInput__input needsclick"
          disabled={false}
          id="endDate"
          name="endDate"
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="End Date"
          readOnly={false}
          required={false}
          type="text"
          value="1995-12-30"
        />
        <p
          className="screen-reader-only"
          id="DateInput__screen-reader-message-endDate"
        >
          Press the down arrow key to interact with the calendar and
            select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
        </p>
        <div
          className="DateInput__display-text DateInput__display-text--has-input"
        >
          12/30/1995
        </div>
      </div>
    </div>
  </div>
</div>
`;
